R/module_nested_tabs.R

```diff
@@ -258,3 +227,38 @@ srv_nested_tabs.teal_module <- function(id, datasets, modules, reporter) {
   }
   reactive(modules)
 }
+
+#' Convert `FilteredData` to reactive list of data
+#'
+#' Converts `FilteredData` object to list of data containing datasets needed for specific module.
+#' Please note that if module needs dataset which has a parent, then parent will be also returned.
+#'
+#' @param module (`teal_module`) module where needed filters are taken from
+#' @param datasets (`FilteredData`) object where needed data are taken from
+#' @return list of reactive datasets with following attributes:
+#' - `code` (`character`) containing datasets reproducible code.
+#' @keywords internal
+#' - `join_keys` (`JoinKeys`) containing relationships between datasets.
+.datasets_to_data <- function(module, datasets) {
+  datanames <- if (identical("all", module$filter)) {
+    datasets$datanames()
+    } else {
+    datasets$get_filterable_datanames(module$filter) # get_filterable_datanames adds parents if present
+    }
```

Comment:
Is there a reason we can't call `datasets$handle_active_datanames(modules$filter)` here?

Reply:
oh yeah I see we have the line `intersect(self$datanames(), datanames)` which means we can't...